array.rb

```diff
@@ -45,15 +45,23 @@ def each
   end
 
   # call-seq:
-  #    array.shuffle!(random: Random) -> array
+  #   shuffle!(random: Random) -> self
   #
-  # Shuffles the elements of +self+ in place.
-  #    a = [1, 2, 3] #=> [1, 2, 3]
-  #    a.shuffle!    #=> [2, 3, 1]
-  #    a             #=> [2, 3, 1]
+  # Shuffles all elements in +self+ into a random order,
+  # as selected by the object given by keyword argument +random+;
+  # returns +self+:
   #
-  # The optional +random+ argument will be used as the random number generator:
-  #    a.shuffle!(random: Random.new(1))  #=> [1, 3, 2]
+  #   a =             [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+  #   a.shuffle! # => [5, 3, 8, 7, 6, 1, 9, 4, 2, 0]
+  #   a.shuffle! # => [9, 4, 0, 6, 2, 8, 1, 5, 3, 7]
+  #
+  #   a =             [0, 1, 0, 1, 0, 1, 0, 1, 0, 1]
+  #   a.shuffle! # => [0, 1, 1, 1, 1, 0, 0, 0, 1, 0]
+  #   a.shuffle! # => [0, 0, 0, 1, 1, 0, 1, 1, 1, 0]
```

Comment:
What's the purpose of the second example here?

Reply:
Removed.

Reply:
What I meant is this whole example. I find that unless an example is obvious, it is confusing to make the reader guess that you're trying to demonstrate with it. Are you trying to demonstrate duplicates here? It's not obvious what it's trying to show here without words to describe it.

Reply:
Clarified(?).